lib/Doctrine/ORM/EntityManager.php

```diff
@@ -349,12 +357,21 @@ public function detach($object)
      * @throws OptimisticLockException If a version check on an entity that
      *         makes use of optimistic locking fails.
      * @throws ORMException
+     * @throws LogicException
      */
     public function flush()
     {
+        if ($this->flushInProgress) {
+            throw new LogicException('EntityManager::flush must not be called inside another EntityManager::flush');
```

Comment:
We will need an entry in `UPGRADE.md`, since this is still a BC break for ORM v3

Reply:
This will need to be a specific exception, not SPL one.
Also please use `()` for the method names.

Reply:
* Added info to `UPGRADE.md`
* Custom exception used. Let me know if we need to rename it or reuse `NotSupported` instead.
* Added `()` to the method defitions